### PR TITLE
Update to new project format

### LINKED ISSUE #230: Investigate migrating to the new .NET project system
https://github.com/dotnet/project-system

### PR DESCRIPTION
**Pull request by scowalt**
- Updated to new dotnet project format
- Updated CI to remove dependency on `nuget` CLI tool
- Removed `OperationAbortedException`

Resolves #230

### CODE REVIEW

**Comment by scowalt** [comment]
I briefly tested this with Bizhawk and StandaloneRTC. Both appear to be working as expected.

File: Source/Frontend/StandaloneRTC/StandaloneRTC.csproj [hunk 1]
```diff
@@ -28,29 +17,24 @@
...
     <UseApplicationTrust>false</UseApplicationTrust>
     <BootstrapperEnabled>true</BootstrapperEnabled>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <UseWindowsForms>true</UseWindowsForms>
+    <!-- Use Build\ instead of Build\net471\ for output -->
+    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
+    <!-- RuntimeIdentifier needed for dolphin -->
+    <RuntimeIdentifiers>win7-x64</RuntimeIdentifiers>
+    <!-- Avoid MissingManifestResourceException  errors, see https://github.com/dotnet/ef6/issues/1258#issuecomment-531355034 -->
+    <EmbeddedResourceUseDependentUponConvention>true</EmbeddedResourceUseDependentUponConvention>
```
Comment (scowalt) [inline Source/Frontend/StandaloneRTC/StandaloneRTC.csproj:27]: This workaround may need to go in more projects, but I don't want to over-include it.

**Comment by NarryG** [comment]
Going to check this one out and test everything to check for any regressions. The documentation said the visual studio experience should be the same but I want to validate there's no designer jank or anything else.

File: Source/Libraries/PluginHost/PluginHost.csproj [hunk 2]
```diff
@@ -1,57 +1,21 @@
...
-    <Optimize>true</Optimize>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
     <OutputPath>..\..\..\Build\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+    <!-- Use Build\ instead of Build\net471\ for output -->
+    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
+    <!-- System.Reflection.Metadata and System.Runtime.CompilerServices.Unsafe are conflicting dependencies https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1605 -->
+    <NoWarn>NU1605</NoWarn>
```
Comment (scowalt) [inline Source/Libraries/PluginHost/PluginHost.csproj:11]: These conflicting dependencies are something we'll probably want to fix eventually. I know that `System.Runtime.CompilerServices.Unsafe` is the one package we can't upgrade at this point (based on our discussion in #129 ), but I'm not sure what the behavior will be when two different versions are downloaded by different parts of the same project.

Comment (NarryG) [reply Source/Libraries/PluginHost/PluginHost.csproj:11]: I'm honestly not sure. I may fork Ceras to resolve the CompilerServices.Unsafe conflict (it wasn't a complicated issue if memory serves me well).  Not sure about the Reflection.Metadata conflict as I don't actually know what's using that.

File: Source/Frontend/StandaloneRTC/StandaloneRTC.csproj (hunk 1 above)
Comment (NarryG) [reply Source/Frontend/StandaloneRTC/StandaloneRTC.csproj:27]: The StandaloneRTC project really shouldn't be included in the SLNs of the other projects so if it's included in Dolphin and preventing it from building without the RuntimeIdentifiers, we should remove that.  

On the second one, could you also add this as part of the comment? https://docs.microsoft.com/en-us/dotnet/core/compatibility/msbuild#resource-manifest-file-name-change

File: Source/Libraries/PluginHost/PluginHost.csproj (hunk 2 above)
Comment (NarryG) [reply Source/Libraries/PluginHost/PluginHost.csproj:11]: I'm honestly not sure. I may fork Ceras to resolve the CompilerServices.Unsafe conflict (it wasn't a complicated issue if memory serves me well). Not sure about the Reflection.Metadata conflict as I don't actually know what's using that.

Comment (scowalt) [reply Source/Libraries/PluginHost/PluginHost.csproj:11]: I opened #244 to investigate this issue. For the purposes of this PR I think we should consider it out-of-scope.

File: Source/Frontend/StandaloneRTC/StandaloneRTC.csproj (hunk 1 above)
Comment (scowalt) [reply Source/Frontend/StandaloneRTC/StandaloneRTC.csproj:32]: I opened #243 to address the first issue, but I think for this PR it's out of scope.

I've updated the comment on the second one to have a bunch of links for context.

**Comment by NarryG** [comment]
When I open ScriptHost I get
```
System.Resources.MissingManifestResourceException: 'Could not find any resources appropriate for the specified culture or the neutral culture.  Make sure "RTCV.NetCore.CloudDebug.resources" was correctly embedded or linked into assembly "NetCore" at compile time, or that all the satellite assemblies required are loadable and fully signed.'```

**Comment by NarryG** [comment]
When I open ScriptHost I get (Advanced tools window > extra tools > script host)
```
System.Resources.MissingManifestResourceException: 'Could not find any resources appropriate for the specified culture or the neutral culture.  Make sure "RTCV.NetCore.CloudDebug.resources" was correctly embedded or linked into assembly "NetCore" at compile time, or that all the satellite assemblies required are loadable and fully signed.  '  
```  

Error is thrown at CloudDebugForm.Designer.CS line 168.  
```
 	mscorlib.dll!System.Resources.ManifestBasedResourceGroveler.HandleResourceStreamMissing(string fileName) Line 687	C#
 	mscorlib.dll!System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(System.Globalization.CultureInfo culture, System.Collections.Generic.Dictionary<string, System.Resources.ResourceSet> localResourceSets, bool tryParents, bool createIfNotExists, ref System.Threading.StackCrawlMark stackMark) Line 160	C#
 	mscorlib.dll!System.Resources.ResourceManager.InternalGetResourceSet(System.Globalization.CultureInfo requestedCulture, bool createIfNotExists, bool tryParents, ref System.Threading.StackCrawlMark stackMark) Line 808	C#
 	mscorlib.dll!System.Resources.ResourceManager.InternalGetResourceSet(System.Globalization.CultureInfo culture, bool createIfNotExists, bool tryParents) Line 752	C#
 	mscorlib.dll!System.Resources.ResourceManager.GetObject(string name, System.Globalization.CultureInfo culture, bool wrapUnmanagedMemStream) Line 1422	C#
>	NetCore.dll!RTCV.NetCore.CloudDebug.InitializeComponent() Line 168	C#
 	NetCore.dll!RTCV.NetCore.CloudDebug.CloudDebug(System.Exception ex, bool canContinue) Line 22	C#
 	StandaloneRTC.exe!StandaloneRTC.Program.ApplicationThreadException(object sender, System.Threading.ThreadExceptionEventArgs e) Line 72	C#
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.OnThreadException(System.Exception t)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Control.WndProcException(System.Exception e)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.NativeWindow.Callback(System.IntPtr hWnd, int msg, System.IntPtr wparam, System.IntPtr lparam)	Unknown
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(System.IntPtr dwComponentID, int reason, int pvLoopData)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(int reason, System.Windows.Forms.ApplicationContext context)	Unknown
 	System.Windows.Forms.dll!System.Windows.Forms.Application.ThreadContext.RunMessageLoop(int reason, System.Windows.Forms.ApplicationContext context)	Unknown
 	StandaloneRTC.exe!StandaloneRTC.Program.StartLoader(string[] args) Line 49	C#
 	StandaloneRTC.exe!StandaloneRTC.Program.Main(string[] args) Line 26	C#
```

**Comment by scowalt** [comment]
Fixed. There's a standard work-around I'm using for `MissingManifestResourceException` errors

**Comment by NarryG** [comment]
Still getting it. Assuming it has to be applied to NetCore as well.

**Comment by scowalt** [comment]
I'm not getting it. Maybe try a clean RTCV build?

**Comment by scowalt** [comment]
I could apply the workaround to all of the projects just in case but I would prefer not to if we can avoid it, just to avoid complexity and unintended side-effects.

**Comment by NarryG** [comment]
Works now. Clean + rebuild fixed it.

Next issue I see is that the TestPlugin folder is building to net471 instead of straight to plugins.

Also, It appears that the "copy local" behavior got lost on a lot of the projects. All three plugin projects are copying references that they didn't previously copy into the output folder. The "Copy Local" box is empty on the projects (CorruptCore, Netcore, PluginHost) so it appears the setting got lost and is defaulting to yes.

**Comment by scowalt** [comment]
I fixed the TestPlugin issue.

I might have fixed the Copy Local issue with `<Private>False</Private>` on the `ProjectReference` objects in the `.csproj` files. However, I wasn't able to find any "Copy Local" box on my Visual Studio. If you're still seeing this issue, could you let me know how to see that box?

**Comment by NarryG** [comment]
Private appears to have fixed it 
image

File: Source/Frontend/StandaloneRTC/StandaloneRTC.csproj [hunk 3]
```diff
@@ -28,29 +17,27 @@
     <IsWebBootstrapper>false</IsWebBootstrapper>
     <UseApplicationTrust>false</UseApplicationTrust>
     <BootstrapperEnabled>true</BootstrapperEnabled>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <UseWindowsForms>true</UseWindowsForms>
```
Comment (scowalt) [inline Source/Frontend/StandaloneRTC/StandaloneRTC.csproj:21]: There might be an issue where form designer doesn't open in projects that have converted to the new format. See https://stackoverflow.com/questions/53487757/how-to-open-form-designer-in-visual-studio

Until I can figure this out, I'm making this change a draft. I consider this a blocking issue.

Comment (scowalt) [reply Source/Frontend/StandaloneRTC/StandaloneRTC.csproj:21]: Nevermind, this is only happening in the RTC-Launcher project

**Comment by NarryG** [comment]
Based on my testing this seems to be good to go. If any regressions show up, we can fix them as we go along.

**Comment by scowalt** [comment]
🥳

**Pull request merged**